site/app/controllers/MiscController.php

```diff
@@ -24,59 +24,110 @@ public function run() {
         }
     }
 
-    private function displayFile() {
-        //Additional security
-        if (!($this->core->getUser()->accessGrading())) {
-            throw new \InvalidArgumentException("It does not look like you're allowed to access this page.");
-        }
-        foreach (explode(DIRECTORY_SEPARATOR, $_REQUEST['path']) as $part) {
-            if ($part == ".." || $part == ".") {
-                throw new \InvalidArgumentException("Cannot have a part of the path just be dots");
+    // function to check that this is a valid access request
+    private function checkValidAccess($is_zip, $dir="",$path="",$user_id="") {
```

Comment:
I would say show an access denied page/popup for all failure conditions. For security type features, it's generally best to be as vague as to what's actually causing the failure for any potential bad actors.